core/keyboard_nav/line_cursor.ts

```diff
@@ -14,120 +14,27 @@
-import {FieldLabel} from '../field_label.js';
-import {FieldNumber} from '../field_number.js';
-import {FieldTextInput} from '../field_textinput.js';
-import {FlyoutButton} from '../flyout_button.js';
-import {FlyoutSeparator} from '../flyout_separator.js';
 import {getFocusManager} from '../focus_manager.js';
+import type {IFocusableNode} from '../interfaces/i_focusable_node.js';
 import {isFocusableNode} from '../interfaces/i_focusable_node.js';
-import type {INavigable} from '../interfaces/i_navigable.js';
 import * as registry from '../registry.js';
-import {RenderedConnection} from '../rendered_connection.js';
-import {Renderer} from '../renderers/zelos/renderer.js';
 import {WorkspaceSvg} from '../workspace_svg.js';
-import {BlockNavigationPolicy} from './block_navigation_policy.js';
-import {ConnectionNavigationPolicy} from './connection_navigation_policy.js';
-import {FieldNavigationPolicy} from './field_navigation_policy.js';
-import {FlyoutButtonNavigationPolicy} from './flyout_button_navigation_policy.js';
-import {FlyoutNavigationPolicy} from './flyout_navigation_policy.js';
-import {FlyoutSeparatorNavigationPolicy} from './flyout_separator_navigation_policy.js';
 import {Marker} from './marker.js';
-import {WorkspaceNavigationPolicy} from './workspace_navigation_policy.js';
-
-/** Options object for LineCursor instances. */
-export interface CursorOptions {
-  /**
-   * Can the cursor visit all stack connections (next/previous), or
-   * (if false) only unconnected next connections?
-   */
-  stackConnections: boolean;
-}
-
-/** Default options for LineCursor instances. */
-const defaultOptions: CursorOptions = {
-  stackConnections: true,
-};
 
 /**
  * Class for a line cursor.
  */
 export class LineCursor extends Marker {
   override type = 'cursor';
 
-  /** Options for this line cursor. */
-  private readonly options: CursorOptions;
-
   /** Locations to try moving the cursor to after a deletion. */
-  private potentialNodes: INavigable<any>[] | null = null;
-
-  /** Whether the renderer is zelos-style. */
-  private isZelos = false;
+  private potentialNodes: IFocusableNode[] | null = null;
 
   /**
    * @param workspace The workspace this cursor belongs to.
-   * @param options Cursor options.
    */
-  constructor(
-    protected readonly workspace: WorkspaceSvg,
-    options?: Partial<CursorOptions>,
-  ) {
+  constructor(protected readonly workspace: WorkspaceSvg) {
     super();
-    // Regularise options and apply defaults.
-    this.options = {...defaultOptions, ...options};
-
-    this.isZelos = workspace.getRenderer() instanceof Renderer;
-
-    this.registerNavigationPolicies();
-  }
-
-  /**
-   * Registers default navigation policies for Blockly's built-in types with
-   * this cursor's workspace.
-   */
-  protected registerNavigationPolicies() {
-    const navigator = this.workspace.getNavigator();
-
-    const blockPolicy = new BlockNavigationPolicy();
-    if (this.workspace.isFlyout) {
```

Comment:
I like the refactor to use a separate navigator for flyout--that seems much cleaner.